server/test/api/references.js

```diff
@@ -0,0 +1,107 @@
+/*global describe, it, beforeEach, process*/
+
+var chai = require('chai');
+var chaiHttp = require('chai-http');
+var expect = chai.expect;
+chai.use(chaiHttp);
+
+if (!global.Promise) {
+  var q = require('q');
+  chai.request.addPromises(q.Promise);
+}
+
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0;
+
+var url = 'https://localhost:8080';
+var user = { username : 'superuser', password : 'superuser', project: 1};
+
+describe('The /references API endpoint', function () {
+  var agent = chai.request.agent(url);
+  var new_reference = {
+    id : 2,
```

Comment:
Does the client normally create IDs?  Why are you sending an ID to the server for a new record?